common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -131,27 +131,30 @@ protected final boolean hasScheduledTasks() {
     }
 
     @Override
-    public  ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
         ObjectUtil.checkNotNull(command, "command");
         ObjectUtil.checkNotNull(unit, "unit");
-        if (delay < 0) {
-            throw new IllegalArgumentException(
-                    String.format("delay: %d (expected: >= 0)", delay));
+        ScheduledFutureTask<Void> task = new ScheduledFutureTask<Void>(
+                this, command, null, ScheduledFutureTask.deadlineNanos(unit.toNanos(delay)));
+        if (delay <= 0) {
+            execute(task);
+            return task;
```

Comment:
This return might be blocked by `execute(task)` and can't return immediately. 
The following can be regarded as "immediate execution" in some sense
```
if (delay <0) {
    delay = 0;
}
return schedule(new ScheduledFutureTask<Void>(
    this, command, null, ScheduledFutureTask.deadlineNanos(unit.toNanos(delay))));
```